File: QB_Challenges/PrepInsta/JAVA/01_Getting_Started/07_Leap_Year/LeapTernary.java
import java.util.Scanner;

public class LeapTernary {

    public static void isLeap() {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the year: ");
        int year = sc.nextInt();
        sc.close();

        int flag = (year % 400 == 0) || (year % 4 == 0 && year % 100 != 0) ? 1 : 0;

        if(flag == 1){
            System.out.println(year + " is a leap year");
        }else{
            System.out.println(year + " is not a leap year");
        }
    
    }

    public static void main(String[] args) {
        isLeap();
    }
    
}

File: QB_Challenges/PrepInsta/01_Getting_Started/07_Leap_Year/CPP/LeapTernary.cpp
#include <iostream>
using namespace std;

void isLeap(){
    int year;
    cout <<"Enter the yearL ";
    cin >>year;

    int flag = (year % 400 == 0) || (year % 4 == 0 && year % 100 != 0) ? 1 : 0;

    if(flag == 1){
        cout <<year <<" is a leap year" <<endl;
    }else{
        cout <<year <<" is not a leap year" <<endl;
    }



}

int main(){
    isLeap();
    return 0;
}